### PR TITLE
Navigation: e2e tests: default to my only existing menu

### PR DESCRIPTION
**Pull request by MaggieCabrera**
## What?

This is part of the effort to re work all the navigation block e2e tests and migration to playwright. This is one of the user stories described in https://github.com/WordPress/gutenberg/issues/45199

This test checks that when only one navigation menu exists, inserting a Navigation block will default correctly to using said menu.

## Testing Instructions

Run npm run test:e2e:playwright -- editor/blocks/navigation.spec.js

### CODE REVIEW

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 1]
```diff
@@ -48,5 +60,89 @@ test.describe(
 <!-- /wp:navigation -->`
 			);
 		} );
+
+		test( 'default to my only existing menu', async ( {
+			editor,
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js:66]: We can also bring out `page` here so that we don't have to do `editor.page` below. We should probably mark `editor.page` and similar variables as private class fields.

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 2]
```diff
@@ -48,5 +60,89 @@ test.describe(
...
+			} );
+
+			await editor.insertBlock( { name: 'core/navigation' } );
+
+			// Check the markup of the block is correct.
+			await editor.publishPost();
+			const content = await editor.getEditedPostContent();
+			expect( content ).toBe(
+				`<!-- wp:navigation {"ref":${ createdMenu.id }} /-->`
+			);
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: Nit: We can also leverage `editor.getBlocks()` as shown [here](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/overusing-snapshots.md#the-solution) for slightly more readable output.

```suggestion
			await expect.poll( editor.getBlocks ).toEqual( [ {
				name: 'core/navigation',
				ref: createdMenu.id,
			} ] );
```

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 3]
```diff
@@ -48,5 +60,89 @@ test.describe(
...
+
+			//check the block in the frontend.
+			await editor.page.goto( '/' );
+			await expect(
+				editor.page.locator(
+					'nav:has-text("WordPress") >> role=link[name="WordPress"]'
+				)
+			).toBeVisible();
+
+			await editor.page.pause();
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: Forgot to delete? 😆

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 4]
```diff
@@ -48,5 +60,89 @@ test.describe(
...
+				`<!-- wp:navigation {"ref":${ createdMenu.id }} /-->`
+			);
+			await editor.page
+				.locator( 'role=button[name="Close panel"i]' )
+				.click();
+
+			//check the block in the canvas.
+			await expect(
+				editor.page.locator(
+					'role=gridcell[name="Custom Link link"] >> a:has-text("WordPress")'
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: Could we use role selector for the second part here? (`a:has-text("WordPress")`)

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 5]
```diff
@@ -48,5 +60,89 @@ test.describe(
...
+				)
+			).toBeVisible();
+
+			//check the block in the frontend.
+			await editor.page.goto( '/' );
+			await expect(
+				editor.page.locator(
+					'nav:has-text("WordPress") >> role=link[name="WordPress"]'
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: Same here, prefer role selectors over CSS selectors whenever possible.

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 6]
```diff
@@ -48,5 +60,89 @@ test.describe(
...
+		this.requestUtils = requestUtils;
+	}
+
+	/**
+	 * Create a navigation menu
+	 *
+	 * @param {Object} menuData navigation menu post data.
+	 * @return {string} Menu content.
+	 */
+	async createNavigationMenu( menuData ) {
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js:119]: We might want to put these utils into `e2e-test-utils-playwright/request-utils` for contributors to leverage. Not something we need to do now though.

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 7]
```diff
@@ -3,19 +3,31 @@
...
+		await use( new NavigationBlockUtils( { page, requestUtils } ) );
+	},
+} );
+
 test.describe(
 	'As a user I want the navigation block to fallback to the best possible default',
 	() => {
 		test.beforeAll( async ( { requestUtils } ) => {
-			await requestUtils.activateTheme( 'emptytheme' );
+			await requestUtils.activateTheme( 'twentytwentythree' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js:17]: Any reason why we want to use TT3 here?

**Review by kevin940726 — COMMENTED** [review]
Nice work! Just some minor suggestions.

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 7 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js:17]: because `emptytheme` won't show the navigation block unless we navigate to the specific post. It seemed like the simpler solution.

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 6 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js:119]: yeah, I was considering this. I followed Dan's lead from https://github.com/WordPress/gutenberg/pull/42695 but a follow up might see them move to the utils package

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 4 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: I don't think the link had a role I could target, but I'll double check!

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 3 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: yes, good catch!!!

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 2 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: thank you, today I learned!

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 7 above)
Comment (kevin940726) [reply test/e2e/specs/editor/blocks/navigation.spec.js:17]: Maybe a comment above will make it easier to read? 🙂

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 2 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: I'm getting this:
```

    - Expected  -  1
    + Received  + 13

      Array [
        Object {
    -     "name": "core/navigation",
    +     "attributes": Object {
    +       "hasIcon": true,
    +       "icon": "handle",
    +       "maxNestingLevel": 5,
    +       "openSubmenusOnClick": false,
    +       "orientation": "horizontal",
    +       "overlayMenu": "mobile",
            "ref": 127,
    +       "showSubmenuIcon": true,
    +     },
    +     "clientId": "ded86be9-a2b7-4318-8e8f-d79089a0eaff",
    +     "innerBlocks": Array [],
    +     "isValid": true,
    +     "name": "core/navigation",
        },
      ]
```

Comment (kevin940726) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: Oh, right. It matches the full block's content. Instead, we should use `toMatch`:

```js
await expect.poll( editor.getBlocks ).toMatch( [ {
	name: 'core/navigation',
	attributes: { ref: createdMenu.id },
} ] );
```

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 4 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: I checked and this is the wrong selector! it's actually targeting the list view on the sidebar, and that uncovered another bug, so thanks for pointing me in this direction

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 2 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: `toMatchObject` did the trick, `toMatch` was expecting a string 👍

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 4 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: Changed with the proper role selector

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 5 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: it's the frontend here so it was a bit trickier, but I think what I did is fine?

**Comment by MaggieCabrera** [comment]
Ok this should be ready for another review, and hope the tests pass.

File: test/e2e/specs/editor/blocks/navigation.spec.js [hunk 8]
```diff
@@ -48,5 +61,88 @@ test.describe(
...
+					attributes: { ref: createdMenu.id },
+				},
+			] );
+			await page.locator( 'role=button[name="Close panel"i]' ).click();
+
+			//check the block in the frontend.
+			await page.goto( '/' );
+			await expect(
+				page.locator(
+					'role=navigation >> nth=1 >> role=link[name="WordPress"]'
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js]: Any reason why we need `nth=1` here?

Comment (kevin940726) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: Nit: In addition, it will be better if we append the `i` flag at the end of the name field to indicate case insensitiveness.

```suggestion
					'role=navigation >> role=link[name="WordPress"i]'
```

Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: the nth is because we have another navigation block in the footer

Comment (kevin940726) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: But don't we only have one nav that has the "WordPress" link? As long as it's only targeting a unique element, it should be fine :).

File: .github/workflows/performance.yml [hunk 9]
```diff
@@ -70,7 +70,7 @@ jobs:
                   WP_MAJOR="${WP_VERSION_ARRAY[0]}.${WP_VERSION_ARRAY[1]}"
                   ./bin/plugin/cli.js perf $GITHUB_SHA debd225d007f4e441ceec80fbd6fa96653f94737 --tests-branch $GITHUB_SHA --wp-version "$WP_MAJOR"
 
-            - uses: actions/github-script@d556feaca394842dc55e4734bf3bb9f685482fa0 # v6.3.3
+            - uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975 # v6.4.0
```
Comment (kevin940726) [inline .github/workflows/performance.yml]: Why is this change necessary in this PR?

**Review by kevin940726 — APPROVED** [review]
LGTM 👍. Just some small suggestions left but they shouldn't be blocking 💯 . Thanks a lot!

File: .github/workflows/performance.yml (hunk 9 above)
Comment (MaggieCabrera) [reply .github/workflows/performance.yml]: I don't get it, it was a [commit](https://github.com/WordPress/gutenberg/pull/47744/commits/22834076b43884f8b0eff2ea3998fa73191e5c5a) by the dependency bot??

Comment (MaggieCabrera) [reply .github/workflows/performance.yml]: it disappeared with a rebase

File: test/e2e/specs/editor/blocks/navigation.spec.js (hunk 8 above)
Comment (MaggieCabrera) [reply test/e2e/specs/editor/blocks/navigation.spec.js]: you are right, I added it because I debugged the selector first without the link text and I was targeting just the nav part of it. Should be good now

**Comment by MaggieCabrera** [comment]
> LGTM 👍. Just some small suggestions left but they shouldn't be blocking 💯 . Thanks a lot!

Thank you so much for your help with this @kevin940726 !! I learned a lot from your review.
